README.md

```diff
@@ -94,6 +94,48 @@ socket.connect('tcp://127.0.0.1:5555');
 
 *(Function, param: Function)*: Closes the socket. Any buffered inbound messages that were not yet received by the application will be discarded. The nanomsg library will try to deliver any outstanding outbound messages for the time specified by `linger`.
 
+## sending and receiving: writable and readable
+
+### socket.send(msg)
+*(Function, param: String or Buffer)*: equivalent to `socket.send()` in [node.zeromq](https://github.com/JustinTulloss/zeromq.node).
+
+```js
+socket.send('hello from nanømsg!');
+```
+
+`send(msg)` is automatically invoked during `Writable` consumption of some other `Readable` stream. In that case a `Writable`'s `pipe()` method can be used to transmit across readable data sources. See [example for more detail](examples/writablepipe.js). The flow of data distributes to endpoint(s) determined by the particular socket type.
+
+```js
+var fs = require('fs');
+var source = fs.createReadStream(__dirname + 'filename.ext');
```

Comment:
Don't we need the path.join() stuff on this?  It works on unix's, but I don't think it's portable to windows?